### ISSUE TITLE
[Types] DefaultVideoTransformDevice is not assignable to type VideoTransformDevice

### ISSUE THREAD

**Issue by afgiel** [aws/amazon-chime-sdk-js#980]
Using `"amazon-chime-sdk-js": "2.3.0"`

I am implementing a `VideoFrameProcessor` to use as a processor with the `DefaultVideoTransformDevice` and getting typescript errors. 

here is a simplified code snippet 
```
const videoTransformDevice = new DefaultVideoTransformDevice(
  logger, 
  videoInputDevice,
  [ new MyCustomVideoFrameProcessor() ],
); 
await meetingSesison.audioVideo.chooseVideoInputDevice(videoTransformDevice);
```
and here is the typescript error
```
   Type 'DefaultVideoTransformDevice' is not assignable to type 'VideoInputDevice'
    Type 'DefaultVideoTransformDevice' is not assignable to type 'VideoTransformDevice'.
      Types of property 'outputMediaStream' are incompatible.
        Type 'MediaStream | null' is not assignable to type 'MediaStream'.
          Type 'null' is not assignable to type 'MediaStream'.
````

**Comment by XHatan** [comment]
Hey, thanks for reporting this, is there any log that I could take a look as well in order to fix this ?

**Comment by afgiel** [comment]
> Hey, thanks for reporting this, is there any log that I could take a look as well in order to fix this ?

this is a build error coming from typescript. what sort of additional logs would be helpful?

**Comment by XHatan** [comment]
This  https://github.com/aws/amazon-chime-sdk-js/blob/master/src/videoframeprocessor/DefaultVideoFrameProcessorPipeline.ts#L18 might be the culprit, would you be able to replace "MediaStream | null = null; "  with "MediaStream" in your node_modules and see if that works ?

**Comment by afgiel** [comment]
I believe both that line and this line from `DefaultVideoTransformDevice` https://github.com/aws/amazon-chime-sdk-js/blob/master/src/videoframeprocessor/DefaultVideoTransformDevice.ts#L38 are the culprits

**Comment by XHatan** [comment]
yeah, thanks for catching this.
Are you able to get around build issue now ?

**Comment by richnew10** [comment]
If the media stream can actually be `null`, then altering that type is the wrong solution.

**Comment by richnew10** [comment]
Fixed by #983. This will be included in the next release.

**Comment by afgiel** [comment]
thank you @richnew10 & @XHatan 🙏

**Comment by vidya-mohan** [comment]
@afgiel released to NPM 2.4.0
